src/components/sidebar/SafeListRemoveDialog/index.tsx

```diff
@@ -24,13 +26,20 @@ const SafeListRemoveDialog = ({
   chainId: string
 }): ReactElement => {
   const dispatch = useAppDispatch()
+  const safeAddress = useSafeAddress()
+  const safeChainId = useChainId()
   const addressBook = useAddressBook()
   const trackingLabel =
     router.pathname === AppRoutes.welcome.accounts ? OVERVIEW_LABELS.login_page : OVERVIEW_LABELS.sidebar
 
   const safe = addressBook?.[address] || address
 
-  const handleConfirm = () => {
+  const handleConfirm = async () => {
+    // When removing the current counterfactual safe, redirect to the accounts page
+    if (safeAddress === address && safeChainId === chainId) {
+      await router.push(AppRoutes.welcome.accounts)
+    }
+    dispatch(removeUndeployedSafe({ chainId, address }))
     dispatch(removeSafe({ chainId, address }))
```

Comment:
Seems a bit strange that regardless of the type of safe we always call both. I would expect something like so:

```
if (isUndeployedSafe) {
    dispatch(removeUndeployedSafe)
} else {
    dispatch(removeSafe)
}
```

I assume inside the dispatch we don't do anything since we can't find an entry in the respective storage?

Reply:
We have to do both because the Safe is both "added" and CF.

Reply:
You are right. In case an undeployed safe is removed while being pinned we need to do both. If its not pinned and removed we only need to remove it from the undeployed storage. Lets keep it as is as long as those dispatch calls don't have unexpected side-effects.